datameta/api/users.py

```diff
@@ -85,56 +85,58 @@ def put(request: Request):
     # Get the targeted user
     target_user = resource_by_id(db, User, user_id)
 
-    if(target_user == None):
+    if target_user is None:
         raise HTTPNotFound() # 404 User ID not found
 
+    has_group_rights = auth_user.group_admin and auth_user.group.uuid == target_user.group.uuid
+    self_changes = auth_user.uuid == target_user.uuid
+
     # First, check, if the user has the rights to perform all the changes they want
 
     # The user has to be site admin to change another users group
-    if group_id != None:
+    if group_id is not None:
         if not auth_user.site_admin:
             raise HTTPForbidden()
 
     # The user has to be site admin to make another user site admin
-    if site_admin != None:
+    if site_admin is not None:
```

Comment:
I can't apply suggestions here, but something like this might be more readable:
```
if (
    site_admin is not None and
    (not auth_user.site_admin or user_is_target)
):
    raise HTTPForbidden()
```

Reply:
+1